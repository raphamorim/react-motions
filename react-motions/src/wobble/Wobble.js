import React from 'react'
import withWobble from './withWobble'

function Wobble({ children }) {
  return (
    <div>
      {withWobble(children)}
    </div>
  )
}

export default Wobble
