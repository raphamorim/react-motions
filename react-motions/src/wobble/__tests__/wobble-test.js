import React from 'react';
import { mount } from 'enzyme';
import Wobble from '../wobble';

describe('Wobble', () => {
  it('should render', () => {
    const wrapper = shallow(<Wobble>Wobble</Wobble>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should receive duration prop', () => {
    const wrapper = mount(<Wobble duration={3}>Wobble</Wobble>);

    expect(wrapper.props().duration).toEqual(3);
  });
  it('should receive fillMode prop', () => {
    const wrapper = mount(<Wobble fillMode="both">Wobble</Wobble>);
    expect(wrapper.props().fillMode).toEqual("both");
  });
  it('should render correctly when infinite', () => {
    const wrapper = mount(<Wobble infinite>Wobble</Wobble>);

    expect(wrapper.props().infinite).toBe(true);
  });
});
