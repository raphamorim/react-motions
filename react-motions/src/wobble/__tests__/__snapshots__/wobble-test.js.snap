// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wobble should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wobble
    duration={1}
    fillMode="both"
    infinite={false}
  >
    Wobble
  </Wobble>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Wobble",
      "style": Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "wobble",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "wobble",
      },
    },
    "ref": null,
    "rendered": "Wobble",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Wobble",
        "style": Object {
          "WebkitAnimationDuration": "1s",
          "WebkitAnimationFillMode": "both",
          "WebkitAnimationName": "wobble",
          "animationDuration": "1s",
          "animationFillMode": "both",
          "animationName": "wobble",
        },
      },
      "ref": null,
      "rendered": "Wobble",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
